plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021-2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.fhir.proxy.plugin;
+
+import ca.uhn.fhir.rest.api.RequestTypeEnum;
+import ca.uhn.fhir.rest.server.servlet.ServletRequestDetails;
+import com.google.fhir.proxy.interfaces.AccessDecision;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.TextUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OpenSRPSyncAccessDecision implements AccessDecision {
+
+	public static final String CARE_TEAM_TAG_URL = "http://smartregister.org/fhir/care-team-tag";
+
+	public static final String LOCATION_TAG_URL = "http://smartregister.org/fhir/location-id";
+
+	public static final String ORGANISATION_TAG_URL = "http://smartregister.org/organisation-tag";
+
+	public static final String SEARCH_PARAM_TAG = "_tag";
+
+	private String applicationId;
+
+	private final List<String> syncStrategy;
+
+	private boolean accessGranted;
+
+	private List<String> careTeamIds;
+
+	private List<String> locationIds;
+
+	private List<String> organizationIds;
+
+	public OpenSRPSyncAccessDecision(String applicationId, boolean accessGranted, List<String> locationIds, List<String> careTeamIds,
+			List<String> organizationIds, List<String> syncStrategy) {
+		this.applicationId = applicationId;
+		this.accessGranted = accessGranted;
+		this.careTeamIds = careTeamIds;
+		this.locationIds = locationIds;
+		this.organizationIds = organizationIds;
+		this.syncStrategy = syncStrategy;
+	}
+
+	@Override
+	public boolean canAccess() {
+		return accessGranted;
+	}
+
+	@Override
+	public void preProcess(ServletRequestDetails servletRequestDetails) {
+		// TODO: Disable access for a user who adds tags to organisations, locations or care teams that they do not have access to
+		//  This does not bar access to anyone who uses their own sync tags to circumvent
+		//  the filter. The aim of this feature based on scoping was to pre-filter the data for the user
+		if (isSyncUrl(servletRequestDetails)) {
+			// This prevents access to a user who has no location/organisation/team assigned to them
+			if (locationIds.size() == 0 && careTeamIds.size() == 0 && organizationIds.size() == 0) {
```

Comment:
I think this check was added for testing purpose. Please correct me if I am wrong!

Reply:
I have added an explanation here https://github.com/opensrp/fhir-access-proxy/pull/9#discussion_r1019030630. The purpose of the `if` block is the same as in the explanation. What do you think?

Reply:
Responded https://github.com/opensrp/fhir-access-proxy/pull/9#discussion_r1019038124